cpp/unreal_plugins/UrdfBot/Source/UrdfParser.cpp

```diff
@@ -0,0 +1,401 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfParser.h"
+
+#include <XmlFile.h>
+#include <XmlNode.h>
+#include <XmlParser.h>
+
+#include "Assert/Assert.h"
+
+UrdfRobotDesc UrdfParser::parse(const std::string& file_name)
+{
+    FXmlFile file;
+    file.LoadFile(UTF8_TO_TCHAR(file_name.c_str()));
+
+    FXmlNode* robot_node = file.GetRootNode();
+    ASSERT(robot_node->GetTag().Equals(TEXT("robot")));
+
+    return parseRobotNode(robot_node);
+}
+
+UrdfGeometryDesc UrdfParser::parseGeometryNode(FXmlNode* geometry_node)
+{
+    UrdfGeometryDesc geometry;
+
+    auto& child_nodes = geometry_node->GetChildrenNodes();
+    ASSERT(child_nodes.Num() == 1);
+    auto& node = child_nodes[0];
+    FString tag = node->GetTag();
```

Comment:
- Let's use `FString&` because that will avoid an unnecessary copy.

Reply:
fixed